### PR TITLE
Select A room

### PR DESCRIPTION
**Pull request by MlulekiN**

### CODE REVIEW

File: Feature/Selecting_A_Room.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    Then User validates that Rooms page is open
+
+    When The user clicks on the *Click to Select Dates* date picker
+    Then User validates calendar is visible
+
+    When User selects the check-in date "<start_date>" and check out date "<end_date>"
+    And The user clicks the calendar button 'Apply'
+    Then User validates that rooms are displayed
+
+    When The User selects an available room with description '2 Bedroom Family Cottage'
```
Comment (benedictnkosi) [inline Feature/Selecting_A_Room.feature]: add the description to the table

File: src/test/java/stepDefinition/SelectARoomSteps.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    WebDriver driver;
+
+    public SelectARoomSteps(TestContext context) {
+        testContext = context;
+        driver = testContext.getWebDriverManager().getDriver();
+    }
+
+    @When("The User selects an available room with description {string}")
+    public void the_User_selects_an_available_room_with_description(String string) {
+        try {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/SelectARoomSteps.java]: remove this try

File: src/test/java/stepDefinition/SelectARoomSteps.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            SeleniumActions.click(chooseRoom(string), driver);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Then("User validates that the selected room is open with description {string}")
+    public void userValidatesThatTheSelectedRoomIsOpenWithDescriptionSpaciousDoubleRoom(String roomOpen) {
+        WebElement roomIsDisplayed = driver.findElement(RoomPage.openRoom(roomOpen));
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/SelectARoomSteps.java]: change the element name. call it OpenRoomButton or something not roomIsDisplayed

Comment (benedictnkosi) [inline src/test/java/stepDefinition/SelectARoomSteps.java]: the element name OpenRoom sounds like an action. methods are named this way, not elements. rename it

**Review by benedictnkosi — CHANGES_REQUESTED** [review]

File: src/test/java/stepDefinition/SelectARoomSteps.java (hunk 3 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/SelectARoomSteps.java]: Done

File: Feature/Selecting_A_Room.feature (hunk 1 above)
Comment (MlulekiN) [reply Feature/Selecting_A_Room.feature]: Done

File: src/test/java/stepDefinition/SelectARoomSteps.java (hunk 2 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/SelectARoomSteps.java]: Done

File: src/test/java/stepDefinition/SelectARoomSteps.java (hunk 3 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/SelectARoomSteps.java]: Done

File: src/test/java/stepDefinition/RoomsAreDisplayedSteps.java [hunk 4]
```diff
@@ -31,18 +31,18 @@ public void userValidatesCalendarIsPresent() {
     }
 
     @When("User selects the check-in date {string} and check out date {string}")
-    public void userSelectsTheStartDateStartdateAndEndDateEnddate(String dateCheck, String dateCheck1) throws InterruptedException {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/RoomsAreDisplayedSteps.java:34]: what is dateCHeck and datecheck1

File: src/test/java/stepDefinition/SelectARoomSteps.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    WebDriver driver;
+
+    public SelectARoomSteps(TestContext context) {
+        testContext = context;
+        driver = testContext.getWebDriverManager().getDriver();
+    }
+
+    @When("The User selects an available room with description {string}")
+    public void the_User_selects_an_available_room_with_description(String string) {
+            SeleniumActions.click(chooseRoom(string), driver);
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/SelectARoomSteps.java]: element name is an action. dont do this

Comment (benedictnkosi) [reply src/test/java/stepDefinition/SelectARoomSteps.java]: check all your elements

File: src/test/java/stepDefinition/SelectARoomSteps.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    TestContext testContext;
+    WebDriver driver;
+
+    public SelectARoomSteps(TestContext context) {
+        testContext = context;
+        driver = testContext.getWebDriverManager().getDriver();
+    }
+
+    @When("The User selects an available room with description {string}")
+    public void the_User_selects_an_available_room_with_description(String string) {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/SelectARoomSteps.java]: what is String string? check all your parameter names

File: src/test/java/stepDefinition/SelectARoomSteps.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            SeleniumActions.click(chooseRoom(string), driver);
+    }
+
+    @Then("User validates that the selected room is open with description {string}")
+    public void userValidatesThatTheSelectedRoomIsOpenWithDescriptionSpaciousDoubleRoom(String roomOpen) {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/SelectARoomSteps.java]: what is Room open, should this not be roomName

**Review by benedictnkosi — CHANGES_REQUESTED** [review]

File: src/test/java/pages/RoomPage.java [hunk 8]
```diff
@@ -17,7 +17,13 @@ public static By calendarButton(String button) {
         return By.xpath("//div[@class='drp-buttons']/button[text()='" + button + "']");
     }
 
-    public static By roomPageVisible = By.xpath("//h2[text()='Rooms']");
+    public static By roomPageIsVisible = By.xpath("//h2[text()='Rooms']");
```
Comment (benedictnkosi) [inline src/test/java/pages/RoomPage.java]: change name

File: src/test/java/pages/RoomPage.java [hunk 9]
```diff
@@ -7,7 +7,7 @@ public class RoomPage {
     public static By calendarTable = By.xpath("//div[@class='calendar-table']");
     public static By invalidDate = By.xpath("(//div[@class='calendar-table']//td[contains(@class,'disabled')])[1]");
     public static By activeDate = By.xpath("//td[contains(@class,'active') or contains(@class,'available')]");
-    public static By roomView = By.xpath("//div[contains(@class,'maghny-gd')]/descendant::h3[@class='room-title']");
+    public static By roomView = By.xpath("//div[contains(@class,'maghny-gd')]");
```
Comment (benedictnkosi) [inline src/test/java/pages/RoomPage.java]: change heading, you can call this a room tile or container

File: src/test/java/pages/RoomPage.java [hunk 10]
```diff
@@ -7,7 +7,7 @@ public class RoomPage {
     public static By calendarTable = By.xpath("//div[@class='calendar-table']");
     public static By invalidDate = By.xpath("(//div[@class='calendar-table']//td[contains(@class,'disabled')])[1]");
```
Comment (benedictnkosi) [inline src/test/java/pages/RoomPage.java]: call this disabled date, invalid is for a particular test case

File: src/test/java/pages/RoomPage.java [hunk 11]
```diff
@@ -17,7 +17,13 @@ public static By calendarButton(String button) {
         return By.xpath("//div[@class='drp-buttons']/button[text()='" + button + "']");
     }
 
-    public static By roomPageVisible = By.xpath("//h2[text()='Rooms']");
+    public static By roomPageIsVisible = By.xpath("//h2[text()='Rooms']");
 
+    public static By clickRoomName(String room) {
+        return By.xpath("//div[@class='maghny-grid']/descendant::div[@class='room-info']//a[text()='" + room + "']");
```
Comment (benedictnkosi) [inline src/test/java/pages/RoomPage.java:22]: this is not clicking anything

File: src/test/java/stepDefinition/RoomsAreDisplayedSteps.java [hunk 12]
```diff
@@ -20,7 +20,7 @@ public RoomsAreDisplayedSteps(TestContext context) {
     }
 
     @And("The user clicks on the *Click to Select Dates* date picker")
-    public void theUserClicksOnTheClickToSelectDatesTab() throws InterruptedException {
+    public void theUserClicksOnTheClickToSelectDatesTab() {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/RoomsAreDisplayedSteps.java]: this is not tab

File: src/test/java/stepDefinition/RoomsAreDisplayedSteps.java [hunk 13]
```diff
@@ -31,33 +31,30 @@ public void userValidatesCalendarIsPresent() {
     }
 
     @When("User selects the check-in date {string} and check out date {string}")
-    public void userSelectsTheStartDateStartdateAndEndDateEnddate(String dateCheck, String dateCheck1) throws InterruptedException {
-        SeleniumActions.click(RoomPage.dateVariable(dateCheck), driver);
-        SeleniumActions.click(RoomPage.dateVariable(dateCheck1), driver);
+    public void userSelectsTheStartDateStartdateAndEndDateEnddate(String dateSelect, String dateSelect1) {
```
Comment (benedictnkosi) [inline src/test/java/stepDefinition/RoomsAreDisplayedSteps.java]: call the parameters checkInDate and CheckOutDate

**Review by benedictnkosi — CHANGES_REQUESTED** [review]

File: src/test/java/stepDefinition/RoomsAreDisplayedSteps.java (hunk 13 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/RoomsAreDisplayedSteps.java]: Done

File: src/test/java/stepDefinition/RoomsAreDisplayedSteps.java (hunk 12 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/RoomsAreDisplayedSteps.java]: Done

File: src/test/java/pages/RoomPage.java (hunk 11 above)
Comment (MlulekiN) [reply src/test/java/pages/RoomPage.java:22]: Done

File: src/test/java/pages/RoomPage.java (hunk 10 above)
Comment (MlulekiN) [reply src/test/java/pages/RoomPage.java]: Done

File: src/test/java/pages/RoomPage.java (hunk 8 above)
Comment (MlulekiN) [reply src/test/java/pages/RoomPage.java]: Done

File: src/test/java/pages/RoomPage.java (hunk 9 above)
Comment (MlulekiN) [reply src/test/java/pages/RoomPage.java]: Done

File: src/test/java/stepDefinition/SelectARoomSteps.java (hunk 7 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/SelectARoomSteps.java]: Done

File: src/test/java/stepDefinition/SelectARoomSteps.java (hunk 6 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/SelectARoomSteps.java]: Done

File: src/test/java/stepDefinition/SelectARoomSteps.java (hunk 5 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/SelectARoomSteps.java]: Done

File: src/test/java/stepDefinition/RoomsAreDisplayedSteps.java (hunk 4 above)
Comment (MlulekiN) [reply src/test/java/stepDefinition/RoomsAreDisplayedSteps.java:34]: Has been changed

**Review by benedictnkosi — APPROVED** [review]

**Pull request merged**